Added font validation during import and loading.

chockenberry opened the pull request:
Added some code that uses the file data to create a font. If that succeeds, then the font is imported.

Also, when fonts are loaded from the Documents folder, the same check is performed. If the file can't be loaded, it's deleted.

manolosavi committed:
xcode 26

manolosavi committed:
fix info titles

chockenberry committed:
Added font validation during import and loading.

manolosavi commented on xFonts/ViewController.m:
```diff
@@ -502,6 +552,15 @@ - (void)documentPicker:(UIDocumentPickerViewController *)controller didPickDocum
 	[self updateNavigation];
+	
+	if (invalidFonts.count != 0) {
+		NSString *message = [NSString stringWithFormat:@"The following fonts could be read and imported: %@", [invalidFonts componentsJoinedByString:@", "]];
```
typo i think?
```suggestion
		NSString *message = [NSString stringWithFormat:@"The following fonts couldn't be read or imported: %@", [invalidFonts componentsJoinedByString:@", "]];
```

chockenberry replied:
Yep, good catch!

chockenberry committed:
Merge branch 'manolo/xcode-26'

chockenberry committed:
Update xFonts/ViewController.m

Co-authored-by: manolo <mnlsv@me.com>

chockenberry committed:
Completely changed how font installation is detected.

• When the FontInfo is created, it now checks if the font is installed using the PostScript name.
• A new entitlement (com.apple.developer.user-fonts) allows the app to use installed fonts. This includes fonts loaded from a mobileconfig file
• The view controllers use isInstalled to show that installation state.
• The checkmarks in the main view now reflect the correct installation state. A default font is use in the list until the font is installed.
• The detail view does not show a sample preview until after the installation is complete.
• Writing out the .mobileconfig is now done non-atomically - the file replacement may be causing profile install problems, and data loss isn't an issue here.
• Removed all the CTFontManager registration that hasn't been working for several OS releases.

chockenberry commented:
Also completely changed how font installation is detected - see details in https://github.com/manolosavi/xFonts/pull/10/commits/469729b240c7cbe56083d614d5285ab315d55b6f. @manolosavi This is going to require a new app entitlement (Target > Signing & Capabilities > + Capability > Fonts > "Use Installed Fonts").

manolosavi commented on xFonts/xFonts.entitlements:
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>com.apple.developer.user-fonts</key>
```
ill have to look into this one to learn more about it, i havent seen it before! [i trust you that its needed i just mean to learn about it for myself]

chockenberry replied:
It's the same entitlement that Tot uses to allow manually installed fonts to be used.

It really should be the default behavior, but Apple is so damn afraid of fonts.

manolosavi replied:
🫠
